Проектная работа 1-й спринт

Timur233 opened the pull request:
- Верстка шаблона
 - Адаптация под hendlebars
 - Проверка ошибок

Timur233 committed:
Проектная работа 1-й спринт
 - Верстка шаблона
 - Адаптация под hendlebars
 - Проверка ошибок

mikebars1995 commented on src/index.html:
```diff
@@ -0,0 +1,33 @@
+                    <ul class="work-card__links pages-list">
+                        {{#each links}}
+                            <li class="pages-list__item">
+                                <a class="pages-list__link" href="{{link}}">
```
**Нужно исправить**

Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту.


**Можно лучше**


Чтобы сделать сайт более доступным для пользователя, нужно стараться придерживаться **семантической верстки**.
Несколько полезных статей на эту тему:
•	[Что такое семантическая вёрстка и зачем она нужна](https://htmlacademy.ru/blog/html/semantics)
•	[Семантика для циников](https://www.youtube.com/watch?v=B5pomLr059U)

mikebars1995 commented on src/pages/messenger/style.scss:
```diff
@@ -0,0 +1,365 @@
+    flex-direction: column;
+    height: 100%;
+    padding: 20px 8px;
+    background-color: #fff;
```
**Можно лучше**

все **цвета** лучше вынести в `css`-переменные и использовать их, чтобы можно было мгновенно понимать, что это за цвет, и можно было бы подкорректировать его в одном месте  и не искать дублирования во всем коде

mikebars1995 commented on src/pages/messenger/index.html:
```diff
@@ -0,0 +1,139 @@
+                                на аукционе за 45000 евро.
+                            </div>
+                            <div class="chat-history__message user-message user-message--left">
+                                <img src="https://site.iskandarov.kz/storage/uploads/2024/07/15/message-image_uid_669513daaa702.jpg"/>
```
**Можно лучше**


Каждой картинке (тегу `img`) нужен информативный атрибут `alt` (например: имя картинки вставлять), так как при ошибке загрузки картинки на сайте подсветится **название** этой картинки (которое в `alt`). Еще это нужно для **плохо видящих** и `незрячих` людей - для них есть специальные средства, которые распознают текст в `alt` и проговаривают его. Подробнее тут https://habr.com/ru/post/341810/

mikebars1995 commented on src/pages/messenger/index.html:
```diff
@@ -0,0 +1,139 @@
+<body>
+
+    <div class="page-wrapper messenger-page__wrapper">
+        <div class="messenger__layout main-layout">
```
**Нужно исправить**

По семантике нужно помещать весь главный контент страницы в тег `main`.

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Посмотрите на *гитхабе* все комментарии к коду (нужно прокрутить вниз страницу там)

Работа проделана огромная


Что сделано отлично:


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов

Что нужно исправить:

* Нужно обернуть список с ссылками в тег `nav` по семантике, так как это навигация по сайту.
* По семантике нужно помещать весь главный контент страницы в тег `main`.

Что можно улучшить:

* все **цвета** лучше вынести в `css`-переменные и использовать их, чтобы можно было мгновенно понимать, что это за цвет, и можно было бы подкорректировать его в одном месте  и не искать дублирования во всем коде
* Каждой картинке (тегу `img`) нужен информативный атрибут `alt` (например: имя картинки вставлять), так как при ошибке загрузки картинки на сайте подсветится **название** этой картинки (которое в `alt`). Еще это нужно для **плохо видящих** и `незрячих` людей - для них есть специальные средства, которые распознают текст в `alt` и проговаривают его. Подробнее тут https://habr.com/ru/post/341810/ 

Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода.

Timur233 committed:
Правки и ошибки
 - Дублирование id на странице смены пароля
 - Неправильный размер иконки на странице чатов
 - Обернул навигацию на главной странице в тег nav
 - Заметил что на этапе верстки забыл сверстать время
 сообщения в чате. До верстал
 - Ко всем изображениям в приложении добавил аттрибут alt
 - Переделал компонент change-avatar, убрал реализацию через
 background, добавил изображения через тег img, и добавил
 аттрибут alt
 - Cтили компонента change-avatar вынес в отдельный файл
 (ранее я выносил в отдельный файл стили только тех компонентов
 которые не имеют прямого отношения к какой-либо из страниц, но
 сейчас решил сделать именно так что бы не дублировать код, далее
 когда появится возможность, стили этого компонента переедут
 в папку самого компонента)
 - Так же поступил с формой авторизации/регистрации, создал
 отдельный файл "auth-form.scss", так как этот код абсолютно идентичен
 и используются более одного раза
 - Заменил все цвета в интерфейсе на переменные
 - На всех страницах проекта добавил тег main

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
